assets/scripts/dark-mode.js

```diff
@@ -0,0 +1,66 @@
+class DarkMode extends HTMLElement {
+  constructor() {
+    super();
+    this.attachShadow({ mode: "open" });
+  }
+
+  connectedCallback() {
+    this.render();
+    this.setDarkMode();
+  }
+
+  setDarkMode() {
+    const isDarkModeOn = localStorage.getItem("isDarkMode");
+
+    if (isDarkModeOn === "1") {
+      document.body.classList.remove("is-light-mode");
+      document.body.classList.add("is-dark-mode");
+    } else {
+      document.body.classList.remove("is-dark-mode");
+      document.body.classList.add("is-light-mode");
+    }
+  }
+
+  toggleDarkMode() {
+    const isDarkModeOn = localStorage.getItem("isDarkMode");
+    if (isDarkModeOn === "1") {
+      localStorage.removeItem("isDarkMode");
+    } else {
+      localStorage.setItem("isDarkMode", "1");
+    }
+  }
+
+  render() {
+    this.shadowRoot.innerHTML = `<button id="mode-toggle" type="button" 
```

Comment:
There's a simpler way to do this. You can project your button in from the Light DOM using a slot render. 

```
render() {
    this.shadowRoot.innerHTML = `<slot name="button"></slot>`;
  }
```

And then in foot.html, return the html with these 2 small changes

```html
  <dark-mode class="l-footer__mode"
    ><button
      slot="button"
      id="mode-toggle"
      type="button"
      class="e__button--toggle e-button--icon">
      <svg
        xmlns="http://www.w3.org/2000/svg"
        fill="none"
        viewBox="0 0 24 24"
        stroke-width="1.5"
        stroke="currentColor"
        class="e-icon e__button__icon e-icon--moon">
        <path
          stroke-linecap="round"
          stroke-linejoin="round"
          d="M21.752 15.002A9.718 9.718 0 0118 15.75c-5.385 0-9.75-4.365-9.75-9.75 0-1.33.266-2.597.748-3.752A9.753 9.753 0 003 11.25C3 16.635 7.365 21 12.75 21a9.753 9.753 0 009.002-5.998z" />
      </svg>
      <span class="is-invisible">Toggle between light and dark mode</span>
    </button></dark-mode
  >
```

Reply:
eg put the positioning class on the web component and the slot attribute on the button